gen/field.go

```diff
@@ -100,12 +103,36 @@ func (f fieldGroupGenerator) DefineStruct(g Generator) error {
 			// optional collections because omitempty doesn't differentiate
 			// between nil and empty collections.
 
+			jsonTags := ""
+			jsonFieldName := f.Name
+
 			if (isStructType(f.Type) || isPrimitiveType(f.Type)) && !f.Required {
-				return fmt.Sprintf("`json:\"%s,omitempty\"`", f.Name)
+				jsonTags += ",omitempty"
+			}
+
+			if f.Required {
+				jsonTags += ",required"
+			}
+
+			goTag := f.Annotations["go.tag"]
+			if goTag != "" {
+				structTag := reflect.StructTag(goTag)
+				jsonTag := structTag.Get("json")
+				if jsonTag != "" {
+					segments := strings.Split(jsonTag, ",")
+
+					// Assign jsonFieldName to be the name in json tag.
+					jsonFieldName = segments[0]
+				}
+			}
+
+			// If the field name is "-" then it means omit, add no tags
+			if jsonFieldName == "-" {
```

Comment:
There are use cases to having a field in a thriftrw struct not serialized / deserialized into JSON.

For example

```
struct RequestBody {
    1: required string userUuid (go.tag = "json:\"-\"" , zanzibar.http.ref = "headers.x-uuid")
    2: required string name 
}
```

We have use cases for specifying that certain fields in the request body are serialized to headers instead. This means we generate our own client that serializes a field into a header and then calls json.Marshal() to serialize other fields into the body. For this to work correctly we need to mark a thrift field as not being json serialized.